### PR TITLE
New and better documentation

### PR DESCRIPTION
**Pull request by arntj**
Based on the docs from the Ruby version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,12 +98,123 @@ public IUser GetCurrentUser(string hash)
...
+```
+
+And this cleaned value is available in the final published result:
+
+```csharp
+public class MyResultPublisher : IResultPublisher
+{
+    public Task Publish<T, TClean>(Result<T, TClean> result)
+    {
+        // result.Control.Value = <User object>
```
Comment (joncloud) [inline README.md]: Technically IUser here.

**Comment by joncloud** [comment]
Looks great to me!

File: README.md (hunk 1 above)
Comment (arntj) [inline README.md]: Thanks, it's fixed.

**Comment by davezych** [comment]
I like it. :+1:

**Comment by Haacked** [comment]
Thanks!

selfie-6
[_GitHub Selfies_](https://github.com/thieman/github-selfies/)

**Pull request merged**